src/tribler/core/components/libtorrent/download_manager/download.py

```diff
@@ -479,16 +479,21 @@ def set_selected_files(self, selected_files=None, prio: int = 4, force: bool = F
             else:
                 self.config.set_selected_files(selected_files)
 
-            torrent_info = get_info_from_handle(self.handle)
-            if not torrent_info or not hasattr(torrent_info, 'files'):
-                self._logger.error("File info not available for torrent %s", hexlify(self.tdef.get_infohash()))
-                return
+            tree = self.tdef.torrent_file_tree
+            total_files = self.tdef.torrent_info.num_files()
 
-            filepriorities = []
-            torrent_storage = torrent_info.files()
-            for index, file_entry in enumerate(torrent_storage):
-                filepriorities.append(prio if index in selected_files or not selected_files else 0)
-            self.set_file_priorities(filepriorities)
+            if not selected_files:
+                selected_files = range(total_files)
```

Comment:
I dislike that our logic is that if "no files are selected to apply the given priority to" then "the given priority is applied to all files". However, I didn't want to change our current functionality in this PR (perhaps in some future PR).